agent/src/agent/cli/streamsets.py

```diff
@@ -27,10 +27,20 @@ def list_():
 
 
 @click.command()
-def add():
-    streamsets.manager.create_streamsets(
-        _prompt_streamsets(StreamSets(_prompt_url(), '', '', ''))
-    )
+@click.option('--url', type=click.STRING)
+@click.option('--username', type=click.STRING, default='admin')
+@click.option('--password', type=click.STRING, default='admin')
+@click.option('--agent-ext-url', type=click.STRING, default='http://anodot-agent')
+def add(url, username, password, agent_ext_url):
+    if url:
+        _validate_streamsets_url(url)
+        _validate_agent_external_url(agent_ext_url)
+        streamsets_ = StreamSets(url, username, password, agent_ext_url)
+        _validate_streamsets(streamsets_)
+    else:
+        streamsets_ = _prompt_streamsets(StreamSets(_prompt_url(), '', '', ''))
+        streamsets_.agent_external_url = _prompt_agent_external_url(streamsets_)
```

Comment:
why don't you prompt it in the _prompt_streamsets() after streamsets validation?

Reply:
because if username/pass validation fails I want to do infinite retry
and if validation of agent_external_url fails i don't want to prompt username and pass again